### ISSUE TITLE
  What do I need to do to use pvnet on other datasets? Thank you very much.

### ISSUE THREAD

**Issue by ameliamlz** [zju3dv/pvnet#105]

**Comment by pengsida** [comment]
I will release a cleaner code recently, including how to use the custom dataset.

**Comment by ameliamlz** [comment]
Great. I will look forward to hearing from you.

**Comment by pengsida** [comment]
For the custom dataset, you could prepare images, corresponding poses and object masks.
We will release the cleaner code in two months.

**Comment by lceck27** [comment]
Do you have any tips on how to create masks and poses (especially poses)? Thank you a lot

**Comment by pengsida** [comment]
For poses,
1. One method is using 2d-3d correspondences. First, define a set of 3d keypoints. Then, annotate the corresponding 2d keypoints in the target image. Use PnP to solve the 6d pose.
2. Another method is using point cloud registration. Use a depth camera to capture the depth. Unproject the depth to obtain the point cloud in the camera frame. Registrate the point cloud with the 3d mode of the object using softwares like CloudCompare, which will compute the 6d pose automatically.

**Comment by ykim104** [comment]
> Do you have any tips on how to create masks and poses (especially poses)? Thank you a lot

I recommend using https://github.com/F2Wang/ObjectDatasetTools 
I was able to make a custom dataset and used pengsida’s FPS function to generate key points. 

I would appreciate if general train script and dataloader were also available - but I’m also working on that and will let you know if it works! :)

**Comment by pengsida** [comment]
1. The vector-field labels will be produced if the keypoint format is right.
2. We will present the general training script recently.

**Comment by ykim104** [comment]
Thank you!!

**Comment by ameliamlz** [comment]
![wrong](https://user-images.githubusercontent.com/47714992/70620507-725fd300-1c52-11ea-8816-d1ab30726a3d.png)
what shoud i do to solve this problem?

**Comment by pengsida** [comment]
You did not load the model properly.

**Comment by ameliamlz** [comment]
i do not change sth,it can run normally before.but it does not work.can you speak more special?

**Comment by pengsida** [comment]
I don't know.

**Comment by pengsida** [comment]
@ameliamlz @ykim104 I release the clean code, including how to train on the custom dataset.
https://github.com/zju3dv/clean-pvnet

**Comment by ameliamlz** [comment]
thank you very much @pengsida
